src/Controllers/Frontend.php

```diff
@@ -187,7 +187,7 @@ public function record(Silex\Application $app, $contenttypeslug, $slug = '')
 
         // Setting the canonical path and the editlink.
         $paths = $app['resources']->getPaths();
-        $app['resources']->setUrl('canonicalurl', sprintf('%s://%s%s', $paths['protocol'], $paths['canonical'], $content->link()));
+        $app['resources']->setUrl('canonicalurl', sprintf('%s%s', $paths['canonical'], $content->link()));
```

Comment:
Nope, because the canonical always drops the 'path' part, because we know it'll give this kind of trouble if we keep it around.